py/selenium/webdriver/firefox/options.py

```diff
@@ -16,81 +16,90 @@
 # under the License.
 
 from selenium.common.exceptions import WebDriverException
+from selenium.webdriver.firefox.firefox_binary import FirefoxBinary
 from selenium.webdriver.firefox.firefox_profile import FirefoxProfile
 
 
+class Log(object):
+    def __init__(self):
+        self.level = None
+
+    def to_capabilities(self):
+        if self.level is not None:
+            return {"log": {"level": self.level}}
+        return {}
+
+
 class Options(object):
+    KEY = "moz:firefoxOptions"
 
     def __init__(self):
-        self._binary_location = ''
+        self._binary = None
         self._profile = None
         self._arguments = []
-        self._extension_files = []
-        self._extensions = []
-        self._firefox_options = {}
+        self.log = Log()
 
     @property
-    def binary_location(self):
-        """
-        Returns the location of the binary otherwise an empty string
-        """
-        return self._binary_location
+    def binary(self):
```

Comment:
Can we have the original method names back with deprecation warnings in addition to this change. These are in Selenium 2 and Selenium 3 is supposed to be purely removing RC, as requested by @shs96c